subprojects/composite-builds/src/main/java/org/gradle/composite/internal/DefaultIncludedBuild.java

```diff
@@ -42,8 +42,6 @@
     private DefaultDependencySubstitutions dependencySubstitutions;
 
     private GradleLauncher gradleLauncher;
-    private SettingsInternal settings;
-    private GradleInternal gradle;
```

Comment:
I agree with making `DefaultIncludedBuild` just a simple configurable value object, and not having access to `GradleLauncher` or `BuildController`. However I'm not sure we can have `IncludedBuildControllers` drive the `BuildController` interface: this needs to be able to manage each part of the lifecycle of a build, through settings, configuration, graph-evaluation and tasks. I think this will require something like `GradleLauncher`, which might be renamed to `BuildLifecycleController`. (With `BuildController` being the thing used to manage complete atomic build invocations.)